lib/client/backend_client.dart

```diff
@@ -20,32 +25,56 @@ class BackendClient {
     } catch (e) {
       print('Error during HTTPrequest $e');
       return 'Error during HTTP request: $e';
-    }
+    } // } finally {
+    //   // JACKSON: The finally block was added due to adding the httpClient rather than the one-off requests
+    //   httpClient.close();
+    // }
   }
 
   Future<dynamic> postData({required String uri, dynamic body}) async {
     try {
       final Uri url = Uri.http(localhost, uri);
-      final response = await http.post(
+      final response = await httpClient.post(
         url,
         headers: headers,
         body: jsonEncode(body),
       );
-      if (response.statusCode == HttpStatus.created) {
+      if (response.statusCode == 201) {
         print('Post executed successfully');
         return jsonDecode(response.body);
       } else {
-        print(
-            'Failed to execute Post Request. Status code: ${response.statusCode}');
+        String errorMessage =
+            'Failed to execute Post Request. Status code: ${response.statusCode}';
+        print(errorMessage);
+        return errorMessage;
       }
     } catch (e) {
       print('Error during HTTP request: $e');
       rethrow;
     }
   }
 
-  void putData(String data) {
-    //remove from current tasks, add to completed tasks list @PutRequest, refactor cubit later
+  Future<dynamic> putData({required String uri, dynamic body}) async {
+    try {
+      final Uri url = Uri.http(localhost, uri);
+      final response = await httpClient.put(
+        url,
+        headers: headers,
+        body: jsonEncode(body),
+      );
+      if (response.statusCode == 200) {
```

Comment:
Please change all lines matching `statusCode ==` to have the `HttpStatus` enum rather than the int literal. I think that makes this one `HttpStatus.ok`

Reply:
done